src/main/java/org/opensearch/flowframework/indices/GlobalContextHandler.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.indices;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.workflow.CreateIndexStep;
+
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX_MAPPING;
+import static org.opensearch.flowframework.workflow.CreateIndexStep.getIndexMappings;
+
+/**
+ * A handler for global context related operations
+ */
+public class GlobalContextHandler {
+    private static final Logger logger = LogManager.getLogger(GlobalContextHandler.class);
+    private final Client client;
+    private final CreateIndexStep createIndexStep;
+
+    /**
+     * constructor
+     * @param client the open search client
+     * @param createIndexStep create index step
+     */
+    public GlobalContextHandler(Client client, CreateIndexStep createIndexStep) {
+        this.client = client;
+        this.createIndexStep = createIndexStep;
+    }
+
+    /**
+     * Get global-context index mapping
+     * @return global-context index mapping
+     * @throws IOException if mapping file cannot be read correctly
+     */
+    public static String getGlobalContextMappings() throws IOException {
+        return getIndexMappings(GLOBAL_CONTEXT_INDEX_MAPPING);
+    }
+
+    private void initGlobalContextIndexIfAbsent(ActionListener<Boolean> listener) {
+        createIndexStep.initIndexIfAbsent(FlowFrameworkIndex.GLOBAL_CONTEXT, listener);
+    }
+
+    /**
+     * add document insert into global context index
+     * @param template the use-case template
+     * @param listener action listener
+     */
+    public void putTemplateToGlobalContext(Template template, ActionListener<IndexResponse> listener) {
```

Comment:
Can we additionally have a method that allows for updating an entry given a document Id (`workflow_id`). This is for use in the `CreateWorkflow` API, which allows for updating a stored use case template. I envision this as a complete replacement of the use case template in the GC entry, keeping the same document ID.

Reply:
I will mention this in the global context followup issue and add this in the next PR. Let's not iterate based on Issues instead of PRs :)